### PR TITLE
CI: Synchronize n-containers to start webtooling bench

### LINKED ISSUE #3952: metrics: sync n-containers to start webtooling benchmark
We observed that the variation in path-length metric increases with the number of running containers because the containers starts their workloads at different times. To reduce such variation we propose to start the workload on all containers at the same time by using a trigger, it is a file that once is found by their respective, launches the workload.

### PR DESCRIPTION
**Pull request by dborquez**
Strenght the webtooling bench by starting the workload at the
same time in all runninng containers.
The script verifies if a trigger-file exists in the filesystem.

Fixes #3952

Signed-off-by: David Esparza <david.esparza.borquez@intel.com>

### CODE REVIEW

**Review by GabyCT — APPROVED** [review]

File: metrics/storage/webtooling.sh [hunk 1]
```diff
@@ -53,6 +63,22 @@ Usage: $0 <count>
 EOF
 }
 
+# script used to launch the workload
+create_start_script() {
+	rm -rf "$src_dir/$start_script"
```
Comment (devimc) [inline metrics/storage/webtooling.sh:68]: I think you can use `rm -f $CMD`, right?

File: metrics/storage/webtooling.sh [hunk 2]
```diff
@@ -53,6 +63,22 @@ Usage: $0 <count>
...
+#!/bin/bash
+mkdir -p ${TESTDIR}
+
+until [ -f $guest_trigger_file ]; do
+	sleep 1
+done
+pushd $file_path
+node dist/cli.js > $file_name
+EOF
+	chmod +x $src_dir/$start_script
```
Comment (devimc) [inline metrics/storage/webtooling.sh:79]: `chmod +x  ${CMD}`  ?

File: metrics/storage/webtooling.sh [hunk 3]
```diff
@@ -53,6 +63,22 @@ Usage: $0 <count>
 EOF
 }
 
+# script used to launch the workload
+create_start_script() {
+	rm -rf "$src_dir/$start_script"
+cat <<EOF >>"$src_dir/$start_script"
```
Comment (devimc) [inline metrics/storage/webtooling.sh:69]: `cat <<EOF >>"${CMD}"`

File: metrics/storage/webtooling.sh [hunk 4]
```diff
@@ -171,6 +199,8 @@ function main() {
 	    wait $pid
 	done
 
+	sleep 1
```
Comment (devimc) [inline metrics/storage/webtooling.sh:202]: why `1` and not `16`? why this is needed?

File: metrics/storage/webtooling.sh [hunk 5]
```diff
@@ -25,7 +25,17 @@ configuration_file="/usr/share/defaults/kata-containers/configuration.toml"
...
+# Options to control the start of the workload using a trigger-file
+dst_dir="/host"
+src_dir="/tmp"
+trigger_file="$RANDOM.txt"
+guest_trigger_file="$dst_dir/$trigger_file"
+host_trigger_file="$src_dir/$trigger_file"
+start_script="webtooling_start.sh"
+
+CMD="$dst_dir/$start_script"
+MOUNT_OPTIONS="type=bind,src=$src_dir,dst=$dst_dir,options=rbind:ro"
```
Comment (devimc) [inline metrics/storage/webtooling.sh:38]: so, you are passing all the tmpfs (/tmp) to the container? that's not cool, can you create a temporal directory instead `mktemp -d`?

**Review by devimc — COMMENTED** [review]
thanks @dborquez - I've left some comments

File: metrics/storage/webtooling.sh (hunk 1 above)
Comment (dborquez) [reply metrics/storage/webtooling.sh:68]: @devimc thank you

File: metrics/storage/webtooling.sh [hunk 6]
```diff
@@ -20,12 +20,26 @@ IMAGE="docker.io/library/local-web-tooling:latest"
...
 file_path="/web-tooling-benchmark"
 file_name="output"
-CMD="mkdir -p ${TESTDIR}; cd $file_path && node dist/cli.js > $file_name"
+
+# Directory where the webtooling results are stored
+TMP_DIR=$(mktemp --tmpdir -d webtool.XXXXXXXXXX)
+
+# Options to control the start of the workload using a trigger-file
+dst_dir="/host"
+src_dir=$(mktemp --tmpdir -d webtool.XXXXXXXXXX)
```
Comment (devimc) [inline metrics/storage/webtooling.sh:34]: thanks, should this directory removed before exit (cleanup)?

**Review by devimc — APPROVED** [review]
thanks @dborquez

**Pull request merged**